### PR TITLE
Chore/improve tests

### PR DESCRIPTION
**Pull request by jonatasrancan**
report

### CODE REVIEW

**Comment by jonatasrancan** [comment]
@nakedsushi I'm still working on this.
I can't add labels or whatever.

So just added `WIP` on the PR name

**Comment by nakedsushi** [comment]
Thanks for the heads up.

**Comment by andreibondarev** [comment]
@jonatasrancan @nakedsushi Maybe we can drop `simplecov` as well to see the coverage %?

**Comment by nakedsushi** [comment]
@andreibondarev I like that suggestion.

**Comment by jonatasrancan** [comment]
@andreibondarev @nakedsushi Do you guys thinks it's worth to try to get at 90% of coverage?

**Comment by nakedsushi** [comment]
I think if we add mores tests in `audio.rb` we can get to 90% w/o too much more work. That's good enough for me.

Another low hanging fruit is `audio_serializer.rb`

**Comment by jonatasrancan** [comment]
@nakedsushi @andreibondarev 90%

90_tests

File: spec/models/audio_spec.rb [hunk 1]
```diff
@@ -1,27 +1,84 @@
...
-  it "is not valid without a title"  do
-    audio = Audio.new(title: nil)
-    expect(audio).to_not be_valid
+    is_expected.to validate_uniqueness_of(:filename)
   end
 
-  it "is not valid when title is too short" do
-    audio = Audio.new(title: 'boo')
-    expect(audio).to_not be_valid
+  describe '#update_metadata' do
```
Comment (jonatasrancan) [inline spec/models/audio_spec.rb]: @nakedsushi I tried to reproduce the shrine behavior with the file here in the test environment.
But without luck, so I just stubed and mocked a bunch of stuff here.

File: spec/models/audio_spec.rb [hunk 2]
```diff
@@ -1,27 +1,84 @@
...
+      end
+    end
+
+    context "when there isn't a flac file" do
+      before do
+        allow(audio).to receive(:file).and_return file_hash
+        allow(File).to receive(:exist?).with(flac.url).and_return false
+      end
+
+      it 'transcodes the file' do
```
Comment (nakedsushi) [inline spec/models/audio_spec.rb]: Is this test named correctly? When there isn't a flac file, it should *not* transcode the file.

File: spec/models/audio_spec.rb [hunk 3]
```diff
@@ -1,27 +1,84 @@
...
+
+        audio.update_metadata
+      end
+    end
   end
 
-  it "is valid with valid attributes" do
-    expect(Audio.new(filename: 'file.wav', title: 'title')).to be_valid
+  describe '.by_filename' do
+    it 'returns audions that matches with the filename' do
```
Comment (nakedsushi) [inline spec/models/audio_spec.rb]: typo: audions => audios ?

File: spec/models/audio_spec.rb (hunk 2 above)
Comment (jonatasrancan) [reply spec/models/audio_spec.rb]: Ops, you are right.

**Pull request merged**